### PR TITLE
fix(skills): fill unsloth Quick Reference and list scraped reference files

### LINKED ISSUE #106780: [Bug]: skill scraper leaves unsloth's Quick Reference empty and two references/ files unlisted
### Summary

The documentation-scraper that generates skills leaves two defects in
`optional-skills/mlops/training/unsloth`, both live on `main` today. Neither is
caught by `tests/skills/test_authoring_standards.py`, which checks frontmatter,
size and `related_skills` but not these.

### 1. Quick Reference is an unfilled placeholder — zero patterns extracted

`optional-skills/mlops/training/unsloth/SKILL.md` still carries the scraper's
placeholder instead of any pattern:

```
## Quick Reference

### Common Patterns

*Quick reference patterns will be added as you use the skill.*
```

```console
$ git show main:optional-skills/mlops/training/unsloth/SKILL.md | grep -c '^\*\*Pattern '
0
```

For contrast, the sibling skill from the same scraper does it right —
`optional-skills/mlops/training/axolotl` has 7 patterns, each a short sentence
plus a snippet.

This looks like the *same* pattern extractor that failed in the opposite
direction on `pytorch-fsdp`, where whole documentation pages were inlined as
pattern text until the 2026-08-08 standards sweep (1c9433897c / 55982159dd)
moved them to `references/common-patterns.md`. Same component, two failure
modes: everything, or nothing.

### 2. `references/` files that are listed nowhere

The directory holds four files, but both `SKILL.md` and `references/index.md`
name only `llms-txt.md`:

```console
$ git ls-tree --name-only main optional-skills/mlops/training/unsloth/references/
index.md  llms-full.md  llms-txt.md  llms.md

$ git show main:optional-skills/mlops/training/unsloth/SKILL.md | grep -oP '(?<=- \*\*)[^*]+\.md'
llms-txt.md
```

So `llms-full.md` (1,094,024 bytes) and `llms.md` (12,586) are unreachable for
an agent reading the skill — it has no way to learn they exist.

They are not redundant copies. `llms-full.md` carries **135 headings that
`llms-txt.md` does not**, including "understanding reward functions", "how grpo
trains a model", "base models" and "epochs". `llms.md` is the upstream
`llms.txt` link index.

### Suggested checks

Two mechanical checks would close both, and fit naturally alongside the
existing authoring-standards suite (and overlap with the linter proposed in
#37352):

1. No `SKILL.md` retains the scraper's placeholder string.
2. For scraper-generated skills, every `references/*.md` on disk appears in
   both the SKILL.md reference list and `references/index.md`.

A caveat on check 2 from trying it: restrict it to scraper-generated skills
(those carrying `automatically generated from official documentation` **and**
having a `references/index.md`). Run across all skills it produced 77 false
positives, because the `- **name.md** - description` bullet form is the
scraper's own and hand-written skills reference their files as prose or links.

A third check worth considering, on **max line length** rather than file size:
the `pytorch-fsdp` failure was a 130,541-character single line, and a
size-based cap alone would flag `research-paper-writing` (106,051 bytes) which
is perfectly healthy at 2,377 lines with a 334-char longest line.

### Environment

Observed on `main` @ 167448edca. Both items verified with the `git show` /
`git ls-tree` commands above against that revision, so no local checkout state
is involved.

### PR DESCRIPTION
**Pull request by KoNit-K**
## What does this PR do?

- Fill the empty scraper placeholder in `optional-skills/mlops/training/unsloth` Quick Reference with short `**Pattern N:**` entries (same shape as sibling `axolotl`). - List previously unlisted `references/llms-full.md` and `references/llms.md` in both `SKILL.md` and `references/index.md`.

## Related Issue

Fixes #106780

## Type of Change

- ✅ Bug fix (non-breaking change that fixes an issue)

## Changes Made

- `references/llms-full.md` — production or test change in this diff
- `references/llms.md` — production or test change in this diff
- `SKILL.md` — production or test change in this diff
- `references/index.md` — production or test change in this diff
- `common-patterns.md` — production or test change in this diff
- `tests/skills/test_authoring_standards.py` — production or test change in this diff

## How to Test

- [x] `scripts/run_tests.sh` on the files in Changes Made — focused verification
- [x] `scripts/run_tests.sh tests/skills/test_authoring_standards.py -q --tb=short`
- [x] Result: 1391 passed

## Evidence

- BEFORE: the focused probe was RED on origin/main before this change
- AFTER: - [x] `scripts/run_tests.sh` on the files in Changes Made — focused verification — 1391 passed
- CONTROL: neighboring paths listed in How to Test still pass

### Code

- ✅ I've read the Contributing Guide
- ✅ My commit messages follow Conventional Commits
- ✅ I searched for existing PRs to make sure this isn't a duplicate
- ✅ My PR contains only changes related to this fix
- ✅ I've run relevant tests locally (see How to Test)
- ✅ I've added tests for my changes
- ✅ I've tested on my platform: macOS

### Documentation & Housekeeping

- ✅ Documentation update: N/A unless noted in Changes Made
- ✅ `cli-config.yaml.example`: N/A
- ✅ `CONTRIBUTING.md` or `AGENTS.md`: N/A
- ✅ Cross-platform impact considered
- ✅ Tool descriptions/schemas: N/A

### CODE REVIEW

**Comment by Enough1122** [comment]
> AI code review — automated review for reference; please use your judgment.

## PR #106787 — fix(skills): fill unsloth Quick Reference and list scraped reference files

**Summary:** Fills the scraper-leftover Quick Reference placeholder in the unsloth skill with 6 verified patterns, lists previously-unlisted `references/*.md` in SKILL.md + index, and adds two repo-wide authoring-standards tests (no QR placeholder; scraper reference files listed).

**Findings (all Non-blocking):**

1. **Non-blocking — new tests are repo-wide sweeps (`_params()` over all skills). Fail-open scoping is thoughtful** (only scraper-marked skills with an index are checked; handwritten skills skipped — avoids the "77-FP sweep"). But any *other* skill currently carrying the QR placeholder will now fail `test_no_scraper_quick_reference_placeholder`. If the suite is green, all are clean; if other placeholders exist this PR must fix or grandfather them. CI will tell — just flagging the blast radius.
2. **Non-blocking — `test_scraper_reference_files_listed` matches by filename substring (`ref.name not in skill`).** A file named `llms.md` substring-matches inside `llms-full.md`/`llms-txt.md` mentions — lenient direction (fewer FPs). Conversely could false-pass, but as a hygiene guard that's fine.
3. **Non-blocking — content check on the 6 unsloth patterns: standard public API usage (`FastLanguageModel.from_pretrained`, `get_peft_model`, `SFTTrainer`, `for_inference`, `save_pretrained_merged/gguf`).** No version-pinned claims that would rot quickly; matches unsloth idioms. The `pytorch-fsdp` index addition (`common-patterns.md`) is a one-line listing fix. Fine.

Hygiene + guardrails, correctly scoped. No blocking issues.

**Comment by KoNit-K** [comment]
Path C recheck of the Enough1122 AI review (all Non-blocking; no CHANGES_REQUESTED, no review threads, MERGEABLE).

Agreed — no code change:
1. Repo-wide `_params()` sweep is the intended blast radius. Local focused `tests/skills/test_authoring_standards.py` is **1391 passed / 0 failed**, so no other SKILL.md still carries the scraper QR placeholder; nothing to fix or grandfather.
2. `test_scraper_reference_files_listed` stays fail-open (scraper-mark + `references/index.md` only) and matches by filename substring. Lenient for `llms.md` vs `llms-full.md` is acceptable as a hygiene guard; current scraped names do not collide. Tightening matchers would be follow-up, not this PR.
3. Unsloth QR uses public API (`FastLanguageModel.from_pretrained`, `get_peft_model`, `SFTTrainer`, `for_inference`, `save_pretrained_merged`/`gguf`). `pytorch-fsdp` `common-patterns.md` is the one-line index listing the new lint required.

Leaving open for maintainer review. Did not push.

**Comment by Enough1122** [comment]
Ack — repo-wide sweep confirmed intentional with 1391 green, fail-open matcher and public-API QR accepted; no-code-change agreed. Closed.

**Comment by KoNit-K** [comment]
Path C recheck after Enough1122 Ack (“no-code-change agreed. Closed.”).

No new actionable review: still no review threads, no CHANGES_REQUESTED, no CI failures (empty rollup). `merge-tree --write-tree HEAD origin/main` vs `2423385c05` is clean. Last comment was the Ack only — nothing to implement.

Independent delta vs current `main` is still unlanded:
- `optional-skills/mlops/training/unsloth/SKILL.md` on `main` still has the scraper QR placeholder (`Quick reference patterns will be added as you use the skill.`).
- This branch fills six `**Pattern N:**` entries and lists `references/llms-full.md` + `references/llms.md` in SKILL.md and `references/index.md`.
- `pytorch-fsdp` `common-patterns.md` index listing and `test_no_scraper_quick_reference_placeholder` / `test_scraper_reference_files_listed` are still unique to this PR.
- #106780 is still open.

Focused suite: `scripts/run_tests.sh tests/skills/test_authoring_standards.py -q --tb=short` — **1391 passed / 0 failed**. Head remains `c2a81bc964`. No commit/push; keeping #106787 open for maintainer review.
